### PR TITLE
WIP: feature/ReadOnly-Validation

### PR DESCRIPTION
**Pull request by youyuanwu**
This is needed for go-swagger read only support https://github.com/go-swagger/go-swagger/pull/2369

This is ready for merge.

### CODE REVIEW

File: values.go [hunk 1]
```diff
@@ -135,6 +136,27 @@ func MaxLength(path, in, data string, maxLength int64) *errors.Validation {
...
+	// data must be of zero value of its type
+	val := reflect.ValueOf(data)
+	if val.IsValid() {
+		if reflect.DeepEqual(reflect.Zero(val.Type()).Interface(), val.Interface()) {
+			return nil
+		}
+	} else {
+		return nil
+	}
+	// WIP: define in errors pkg and provide good msg
```
Comment (youyuanwu) [inline values.go:156]: Let me know if this PR is ok and I will add error in errors pkg.

**Review by casualjim — APPROVED** [review]

**Comment by casualjim** [comment]
Thanks! this looks great

File: go.mod [hunk 2]
```diff
@@ -5,7 +5,7 @@ go 1.14
 require (
 	github.com/asaskevich/govalidator v0.0.0-20200428143746-21a406dcc535 // indirect
 	github.com/go-openapi/analysis v0.19.10
-	github.com/go-openapi/errors v0.19.6
+	github.com/go-openapi/errors v0.19.7-0.20200915043824-00a6ebeed93c
```
Comment (youyuanwu) [inline go.mod:8]: We need to properly update version. Once errors pkg tag is created.

File: values.go (hunk 1 above)
Comment (youyuanwu) [reply values.go]: This is fixed.

File: go.mod (hunk 2 above)
Comment (casualjim) [reply go.mod:8]: it's tagged you can get it with ```go get github.com/go-openapi/errors@v0.19.7```

**Review by casualjim — APPROVED** [review]

File: go.mod (hunk 2 above)
Comment (youyuanwu) [reply go.mod]: Updated. Thanks.

**Pull request merged**